libmcount/record.c

```diff
@@ -416,6 +416,7 @@ static unsigned save_to_argbuf(void *argbuf, struct list_head *args_spec,
 		if (is_retval != (spec->idx == RETVAL_IDX))
 			continue;
 
+		update_mem_regions(ctx->regions);
```

Comment:
No need to do it here since `mcount_arch_get_arg/retval()` will call it if needed.  And I don't want to add the overhead of doing that unconditionally.